audio2tonie.sh

```diff
@@ -1,96 +1,231 @@
+    python3 "$OPUS_2_TONIE_PATH/opus2tonie.py" "$input" "$output" || { echo "Transcoding failed"; exit 1; }
+}
+
+# Function to upload file to Teddycloud
+upload_to_teddycloud() {
+    local file="$1"
+    local uri="$2"
+    local path="${3:-/}"  # Default to root path if not specified
+
+    echo -n "Uploading file to Teddycloud ($uri) at path '$path'..."
+
+    # Use curl with -L to follow redirects (e.g., HTTP to HTTPS)
+    response_code=$(curl -s -o /dev/null -w "%{http_code}" -L -F "file=@$file" "${uri}/api/fileUpload?path=${path}&special=library") || {
+        echo "Failed to upload file."
+        exit 1
+    }
+
+    if [[ "$response_code" == 200 ]]; then
+        echo ": OK"
+    else
+        echo "Error! Upload failed with HTTP code: $response_code"
+        exit 1
+    fi
+}
+
+# Function to validate and normalize the Teddycloud URI
+normalize_teddycloud_uri() {
+    local uri="$1"
+
+    # Ensure the URI starts with http:// or https://
+    if [[ ! "$uri" =~ ^https?:// ]]; then
+        echo "Error: Teddycloud URI must start with http:// or https://"
+        exit 1
+    fi
+
+    # Remove trailing slashes
+    uri="${uri%/}"
+
+    echo "$uri"
+}
+
+# Function to print TAF file creation details
+print_taf_creation_details() {
+    local input="$1"
+    local output="$2"
+    local count="$3"
+
+    echo "$SEPARATOR"
+    echo "Creating TAF file:"
+    echo "  - Input: $input"
+    echo "  - Output: $output"
+    echo "  - Number of chapters: $count"
+}
+
+# Function to check if a directory contains valid audio files
+has_valid_audio_files() {
+    local dir="$1"
+    local count
+    count=$(find "$dir" -maxdepth 1 -type f \( -name "*.mp3" -o -name "*.mp2" -o -name "*.m4a" -o -name "*.m4b" -o -name "*.opus" -o -name "*.ogg" -o -name "*.wav" -o -name "*.aac" -o -name "*.mp4" \) | wc -l)
+    [[ "$count" -gt 0 ]]
+}
+
+# Function to get the base name of a directory (without trailing slash)
+get_basename() {
+    local path="$1"
+    path="${path%/}"  # Remove trailing slash
+    basename "$path"
+}
+
+# Function to process directories recursively
+process_recursive() {
+    for d in "$SOURCE"/*/; do
+        local dirname
+        dirname=$(basename "$d")
+        OUTPUT_FILE="${dirname}.taf"
+        echo "Current folder: $dirname"
+        local count
+        local count
+        count=$(find "$d" -maxdepth 1 -type f | wc -l)
+        transcode_files "$d" "$SOURCE/$OUTPUT_FILE" "$count"
+        if [[ -n "${TEDDYCLOUD_IP:-}" ]]; then
+            upload_to_teddycloud "$SOURCE/$OUTPUT_FILE"
+        fi
+        echo "$SEPARATOR"
+    done
+    echo "Finished! Enjoy."
+    exit 0
+}
+
+# Main script logic
+main() {
+    echo "$SEPARATOR"
+    print_settings
+
+    if [[ "$SOURCE" == *.lst ]]; then
+        count=$(sed -n '$=' "$SOURCE")
+    elif [[ "$SOURCE" == *"www.ardaudiothek.de"* ]]; then
+        download_ard_episode
+        count=1
+    elif [[ "$SOURCE" == *.* ]]; then
```

Comment:
Nice cleanups!

I noticed this incorrectly handles directories containing a `.`. For example relative directories which are commonly passed with the notation `./folder`. in that case, the script prints the wrong chapter count (1). It's just a display issue, the resulting TAF file contains the correct number of chapters.

I think it would be better to check if `$SOURCE` is a directory or a file instead of relying on wildcard matching.